src/main/java/it/gov/pagopa/pu/migration/connector/fileshare/FileShareServiceImpl.java

```diff
@@ -16,7 +19,21 @@ public FileShareServiceImpl(FileShareClient client) {
 
   @Override
   public Long uploadIngestionFlowFile(Long organizationId, IngestionFlowFileType ingestionFlowFileType, Resource file, String accessToken) {
-    return client.uploadIngestionFlowFile(organizationId, ingestionFlowFileType, file, accessToken);
+    try {
+      return client.uploadIngestionFlowFile(organizationId, ingestionFlowFileType, file, accessToken);
+    } catch (HttpClientErrorException e) {
+      if (e.getStatusCode() == HttpStatus.BAD_REQUEST) {
+        try {
+          ErrorDTO error = e.getResponseBodyAs(ErrorDTO.class);
```

Comment:
it would be better to use it.gov.pagopa.pu.fileshare.dto.generated.FileshareErrorDTO instead